### PR TITLE
Develop rebase

### PR DESCRIPTION
**Pull request by ehildenb**
Cleaned up commits from branch `develop` significantly.

Many squashes, many commits removed which were just reverting other changes, better commit messages.

### CODE REVIEW

**Comment by ehildenb** [comment]
@cos @msaxena2

File: build.sbt [hunk 1]
```diff
@@ -1,16 +0,0 @@
-
-organization := "org.kframework.k"
-
-name := "kore"
-
-scalaVersion := "2.12.2"
```
Comment (cos) [inline build.sbt]: have we decided that we are using maven for building kore?

Comment (ehildenb) [inline build.sbt]: I'll put `build.sbt` into a new branch where we can make sure it works on everyone's system before merging it into master.

**Comment by grosu** [comment]
Is there any reason we do not merge this PR?  I'd like to have only one branch in this repo asap, because we'll start implementing the new K backend in Haskell very soon.  Let's please merge all these PRs asap.

**Comment by msaxena2** [comment]
@grosu These changes were needed as a part of "extended kore" and used mostly in the scala backend. The Haskell backend won't be needing these changes directly, hence they're no longer relevant. For a cleaner Kore repository, I'm closing this PR and deleting the branch. If needed, we can restore the branch later, mainly to look for ideas in this PR.

**Pull request closed without merging**